### ISSUE TITLE
Edit category modal

### ISSUE THREAD

**Issue by JulianKniephoff** [opencast/annotation-tool#455]
Currently categories are edited "in place" by checking an "edit mode" checkbox. This is a bit clumsy, and with new features like #407 and #414 the interface starts to look a bit bloated.

The idea here is to have an editing modal for categories, similar to the one for tracks.

**Comment by JulianKniephoff** [comment]
There is some general discussion about this in #396 already, but we should probably summarize it here and specify exactly what should and what shouldn't be in the new modal.

@dagraf would you be willing to do that in a comment here?

**Comment by JulianKniephoff** [comment]
This should probably be deferred until after #414 and #407 are merged.

**Comment by dagraf** [comment]
> There is some general discussion about this in #396 already, but we should probably summarize it here and specify exactly what should and what shouldn't be in the new modal.
> 
> @dagraf would you be willing to do that in a comment here?

Yes, I am willing to do this here. :) But it has to wait till beginning of next week (at latest).

**Comment by JulianKniephoff** [comment]
See also maybe #373? :thinking:

**Comment by dagraf** [comment]
Short update: I'm working on the specs and will post them here this week.

**Comment by dagraf** [comment]
I suggest creating 4 different modals (create category, edit catgory, create scale, edit scale) and a new 'action-dropdown' in the Public- and Mine-Tab. See mockups and specs for details. I think we should discuss about this in a meeting. What do you think? Please send me some possible dates we could meet online.
[Mockup_Specs Kateogrie-Modal.pdf](https://github.com/opencast/annotation-tool/files/6038905/Mockup_Specs.Kateogrie-Modal.pdf)

**Comment by dagraf** [comment]
See new proposals for... 
- ... how to ged rid of the checkbox "Edit"
- ... deal with missing values in mandatory boxes
- ... deal with the "scales are not possible with series-cateogries"-issue

[Mockup Specs_Kategorie und Skala Modal_v3.pdf](https://github.com/opencast/annotation-tool/files/6297987/Mockup.Specs_Kategorie.und.Skala.Modal_v3.pdf)

**Comment by dagraf** [comment]
See slightly altered Mockups.
[Mockup Specs_Kategorie und Skala Modal_v3.pdf](https://github.com/opencast/annotation-tool/files/6406853/Mockup.Specs_Kategorie.und.Skala.Modal_v3.pdf)

**Comment by JulianKniephoff** [comment]
As far as I can tell this is basically done now. Of course we can always improve things further, but the feature in its basic form has landed.
